src/System.Private.CoreLib/src/System/Runtime/Loader/AssemblyLoadContext.CoreCLR.cs

```diff
@@ -206,15 +206,20 @@ public void StartProfileOptimization(string profile)
         /// </summary>
         private static void StartAssemblyLoad(ref Guid activityId, ref Guid relatedActivityId)
         {
-            ActivityTracker.Instance.OnStart(NativeRuntimeEventSource.Log.Name, AssemblyLoadName, 0, ref activityId, ref relatedActivityId, EventActivityOptions.Recursive);
+            // Make sure ActivityTracker is enabled
+            ActivityTracker.Instance.Enable();
+
+            // Don't use trace to TPL event source in ActivityTracker, as that could lead to recursive loading.
```

Comment:
Is it the right thing to depend on this ActivityTracker for assembly loads? Having the exact right part carved out of it looks fragile.

Reply:
The comment was a bit misguided and I think the comment makes it sound more problematic than it is, as there wasn't actually any recursive loading. The issue was specific to the way the TplEventSource is a singleton - if it triggered a load during its instantiation, the ActivityTracker would then try to use the TplEventSource singleton which wasn't done being instantiated.